### PR TITLE
Consolidate IProject.FilePath and .Directory

### PR DESCRIPTION
**Pull request by twsouthwick**
We don't need separate properties for these. This change replaces these two properties with a single `IProject.FileInfo` property.

### CODE REVIEW

File: src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.cs [hunk 1]
```diff
@@ -21,13 +21,13 @@ internal partial class MSBuildProject : IProject
 
         public MSBuildWorkspaceUpgradeContext Context { get; }
 
-        public string FilePath { get; }
+        public FileInfo FileInfo { get; }
 
-        public string Directory => Path.GetDirectoryName(FilePath)!;
+        public string Directory => FileInfo.Directory!.FullName;
```
Comment (mjrousos) [inline src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.cs:26]: FileInfo.DirectoryName instead, perhaps?

File: src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.ConfigUpdaters/AppSettingsConfigUpdater.cs [hunk 2]
```diff
@@ -51,7 +51,7 @@ public async Task<bool> ApplyAsync(IUpgradeContext context, ImmutableArray<Confi
             var project = context.CurrentProject.Required();
 
             // Determine where appsettings.json should live
-            var appSettingsPath = Path.Combine(project.Directory ?? string.Empty, AppSettingsJsonFileName);
+            var appSettingsPath = Path.Combine(project.FileInfo.DirectoryName, AppSettingsJsonFileName);
```
Comment (mjrousos) [inline src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.ConfigUpdaters/AppSettingsConfigUpdater.cs:54]: DirectoryName can be null, I think, so we still need the ?? here and in the other similar places it was removed.

**Review by mjrousos — APPROVED** [review]
LGTM; left a couple nit-picky comments

File: src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.ConfigUpdaters/AppSettingsConfigUpdater.cs (hunk 2 above)
Comment (twsouthwick) [reply src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.ConfigUpdaters/AppSettingsConfigUpdater.cs:54]: There was no nullable error, so I assumed it was ok... we're targeting .net standard 2.0 which doesn't have those marked

**Pull request merged**